File: Problem_Solving/Algorithms/Warmup/9_Birthday_Cake_Candles/Solution.py
import math
import os
import random
import re
import sys

def birthdayCakeCandles(ar):
    maxVal=0
    candlesQty=0
    for key,val in enumerate(ar):
        if val>maxVal:
            maxVal=val
            candlesQty=1
        elif val==maxVal:
            candlesQty+=1
    return candlesQty

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    ar_count = int(input())

    ar = list(map(int, input().rstrip().split()))

    result = birthdayCakeCandles(ar)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Problem_Solving/Algorithms/Warmup/9_Birthday_Cake_Candles/Solution.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    static int birthdayCakeCandles(int[] ar) {
        int maxVal=0;
        int candlesQty=0;
        foreach(int val in ar){
            if(val>maxVal){
                maxVal=val;
                candlesQty=1;
            }
            else if(val==maxVal)
                candlesQty++;
        }
        return candlesQty;
    }

    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int arCount = Convert.ToInt32(Console.ReadLine());

        int[] ar = Array.ConvertAll(Console.ReadLine().Split(' '), arTemp => Convert.ToInt32(arTemp));
        int result = birthdayCakeCandles(ar);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}
